### PR TITLE
Added message where there are problems with the network

### PR DESCRIPTION
**Pull request by teamagam3**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/42" target="_blank">ReviewNinja

File: app/src/main/java/com/teamagam/gimelgimel/app/network/receivers/ConnectivityStatusReceiver.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * A receiver that is notified for any connectivity problem and changes.
+ * The receiver uses {@link NetworkAvailableListener} to notify other classes about any changes in the network.
+ */
+public class ConnectivityStatusReceiver extends BroadcastReceiver {
+
+    public static final String INTENT_NAME = "connectivityStatus";
+    public static final String NETWORK_AVAILABLE_EXTRA = "networkAvailable";
+
+    private NetworkAvailableListener mNetworkAvailableListener;
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/network/receivers/ConnectivityStatusReceiver.java:39]: If there's no point in instantiating this object without a listener, i'd say you should ask for one at the constructor

File: app/src/main/java/com/teamagam/gimelgimel/app/network/receivers/NetworkChangeReceiver.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import okhttp3.Request;
+
+/**
+ * An implementation of {@link BroadcastReceiver}, that gets notification every time the network changes.
+ */
+public class NetworkChangeReceiver extends BroadcastReceiver {
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        final ConnectivityManager connectivityManager =
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/network/receivers/NetworkChangeReceiver.java:47]: Extract two methods out of this method.
isNetwrokAvailbale
broadcastConnectivityStatus !resolve

File: app/src/main/java/com/teamagam/gimelgimel/app/network/services/GGImageService.java [hunk 3]
```diff
@@ -50,6 +52,8 @@ protected void onHandleIntent(Intent intent) {
         long time = (long) extras.get(IImageSender.TIME);
         PointGeometry loc = (PointGeometry) extras.get(IImageSender.LOCATION);
 
+        LocalBroadcastManager.getInstance(this).sendBroadcast(new Intent(ConnectivityStatusReceiver.INTENT_NAME));
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/network/services/GGImageService.java]: remove !resolve

File: app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java [hunk 4]
```diff
@@ -166,6 +173,13 @@ public void run() {
                 });
             }
         }, Message.IMAGE);
+
+        mConnectivityStatusReceiver = new ConnectivityStatusReceiver();
+        mConnectivityStatusReceiver.setListener(this);
+
+        IntentFilter intentFilter = new IntentFilter(ConnectivityStatusReceiver.INTENT_NAME);
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java:179]: Double registration of receiver.
registration should be done on onResume alone !resolve

File: app/src/main/java/com/teamagam/gimelgimel/app/network/receivers/NetworkChangeReceiver.java (hunk 2 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/network/receivers/NetworkChangeReceiver.java:47]: !resolved

File: app/src/main/java/com/teamagam/gimelgimel/app/network/services/GGImageService.java (hunk 3 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/network/services/GGImageService.java]: !resolved

File: app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java (hunk 4 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java:179]: !resolved

File: app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java [hunk 5]
```diff
@@ -166,6 +173,12 @@ public void run() {
                 });
             }
         }, Message.IMAGE);
+
+        mConnectivityStatusReceiver = new ConnectivityStatusReceiver(this);
+
+        IntentFilter intentFilter = new IntentFilter(ConnectivityStatusReceiver.INTENT_NAME);
+
+        LocalBroadcastManager.getInstance(this).registerReceiver(mConnectivityStatusReceiver, intentFilter);
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java]: registration should be done only on onResume .
There's double registration there. !resolve

File: app/src/main/java/com/teamagam/gimelgimel/app/network/receivers/ConnectivityStatusReceiver.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    public interface NetworkAvailableListener {
+        void onNetworkAvailableChange(boolean isNetworkAvailable);
+    }
+
+    /**
+     * Broadcasts a new status of the network to {@link ConnectivityStatusReceiver}
+     * @param context The context to use
+     * @param isNetworkAvailable True if we have network connection, else false
+     */
+    public static void sendBroadcast(Context context, boolean isNetworkAvailable) {
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/network/receivers/ConnectivityStatusReceiver.java]: Split into 2 methods that doesn't take boolean argument.
move static method declaration to the top of the file/
sendBroadcast is redundant, broadcast suffice. !resolve

Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/network/receivers/ConnectivityStatusReceiver.java]: !resolved

File: app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java (hunk 5 above)
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java]: !resolved

**Pull request merged**